okhttp/src/main/java/okhttp3/Challenge.java

```diff
@@ -16,67 +16,92 @@
 package okhttp3;
 
 import java.nio.charset.Charset;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
 import javax.annotation.Nullable;
 
+import static java.util.Collections.singletonMap;
+import static java.util.Collections.unmodifiableMap;
 import static okhttp3.internal.Util.ISO_8859_1;
 
 /** An RFC 7617 challenge. */
 public final class Challenge {
   private final String scheme;
-  private final String realm;
-  private final Charset charset;
+  private final Map<String, String> authParams;
 
-  public Challenge(String scheme, String realm) {
-    this(scheme, realm, ISO_8859_1);
+  public Challenge(String scheme, Map<String, String> authParams) {
+    if (scheme == null) throw new NullPointerException("scheme == null");
+    if (authParams == null) throw new NullPointerException("authParams == null");
+    this.scheme = scheme.toLowerCase();
```

Comment:
I don’t think this is worthwhile, and besides it won’t work in Turkey!
https://en.wikipedia.org/wiki/Dotted_and_dotless_I

Reply:
What do you mean?
The `toLowerCase`?
I don't think there will be auth schemes or auth param names in turkey, do you?
I personally would that accept that as case where one then would have to parse the challenge header themselves.

Reply:
The issue is that if you have an Android device configured with the default locale as Turkey then "BASIC".toLowerCase() returns "basıc". So you should never use that method without passing a parameter of Locale.US.

Reply:
https://developer.android.com/reference/java/util/Locale#default_locale

Reply:
Oh, ok, I wasn't aware of that.
Escpecially as it doesn't happen here if I use turkish locale for the `toLowerCase` call.
I added the explicit US locale. now

Reply:
TIL!